### PR TITLE
Feat(eos_cli_config_gen): Add capability to hide passwords and keys in generated doc and conf

### LINKED ISSUE #2493: Remove passwords, hashes from documentation (sanitized)
### Enhancement summary

Currently our documentation includes passwords, hashes etc by default. The current workaround is to use .gitignore to avoid uploading that information in the repository.

We are looking for an option to generate the documentation but avoid including any passwords or hashes (using asterisks would be fine). Similar to our "show run sanitized" option that removes any sensitive data in the running-config.

Some of the commands to be sanitized are:
- snmp-communites 
- snmp-groups
- snmp-users
- bgp passwords
- local users
- enable password

```
snmp-server community <removed> ro
snmp-server user test group v3 localized f5717f00000000aaa100 auth sha <removed> priv aes <removed>
snmp-server host 1.2.3.4 version 2c <removed>
snmp-server host 10.81.120.60 vrf mgmt version 2c <removed>

router bgp 65000.1
   neighbor 1.1.1.1 password 7 <removed>
   neighbor 1.2.3.4 password 7 <removed>
   neighbor peer-group-test password 7 <removed>

username cvpadmin privilege 15 role network-admin secret sha512 <removed>. 
enable password sha512 <removed>
```

### Which component of AVD is impacted

others

### Use case example

Avoid uploading sensitive information in the repository.

### Describe the solution you would like

Option to sanitize documentation.

### Describe alternatives you have considered

use .gitignore to avoid this.

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Introduce a new filter `hide_passwords` which allow to replace a value in a Jinja Template by `<removed>`.
By default the values identified in Jinja are hidden in the documentation but kept in the generated config (cf `eos_cil_config_gen/default/main/main.yml`)

## Related Issue(s)

Fixes #2493

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

* new filter implemented
* added default values for documentation (`true`) and config (`false`)

## How to test

molecule documentation updated.

### User Checklist

- [x] Add tests for filter
- [x] Add molecule test for config filter
- [x] plugin documentation written

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/aaa.md [hunk 1]
```diff
@@ -58,8 +58,8 @@ interface Management1
 ```eos
 !
 username admin privilege 15 role network-admin nopassword
-username ansible privilege 15 role network-admin secret sha512 $6$.I7/ZR/zlLIUv8fr$vR/JvLTbq5amMt6Y1SE4CKlPDv/AzJYlFYHkUZ17BDovm0Oi4aLdBULe1EmZ0Y9xKjVLMKpxCSKmlrAioDxbQ0
-username cvpadmin privilege 15 role network-admin secret sha512 $6$.I7/ZR/zlLIUv8fr$vR/JvLTbq5amMt6Y1SE4CKlPDv/AzJYlFYHkUZ17BDovm0Oi4aLdBULe1EmZ0Y9xKjVLMKpxCSKmlrAioDxbQ0
+username ansible privilege 15 role network-admin secret sha512 <removed>
+username cvpadmin privilege 15 role network-admin secret sha512 <removed>
 username cvpadmin ssh-key ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC9OuVC4D+ARBrc9sP0VRmP6osTo8fgA4Z/dkacQuiOgph6VTHaBkIuqR7XswKKCOH36GXeIChnIF+d1HSoe05mZX+bT2Nu1SObnO8jZjqIFZqUlXUTHWgmnChchABmXS3KMQlivVDE/r9o3vmHEFTfKPZsmG7YHZuavfYXxFJtqtDW0nGH/WJ+mm4v2CP1tOPBLvNE3mLXXyTepDkmrCH/fkwgPR3gBqLrkhWlma0bz+7I851RpCQemhVJFxeI/SnvQfL2VJU2ZMM3pPRSTlLry7Od6kZNAkr4dIOFDCVAaIDbBxPUZ/LvPfyEUwicEo/EKmpLBQ6E2UqcCK2pTyV/K63682spi2mkxp4FgaLi4CjWkpnL1A/MD7WhrSNgqXToF7QCb9Lidagy9IHafQxfu7LwkFdyQIMu8XNwDZIycuf29wHbDdz1N+YNVK8zwyNAbMOeKMqblsEm2YIorgjzQX1m9+/rJeFBKz77PSgeMp/Rc3txFVuSmFmeTy3aMkU= cvpadmin@hostmachine.local
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/aaa.md:63]: I don't know if a ssh-key is considered dangerous. I think we should remove it.

**Review by ClausHolbechArista — COMMENTED** [review]
I found a few more:
- daemon_terminattr the ingest auth key
- radius_server keys (seen in deprecated vars so maybe old variant?)
- snmp community with vrf and version
- documentation tables with snmp communities. That becomes a bit hard...

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/aaa.md (hunk 1 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/aaa.md:63]: Fixing all the one in next commit. 
As discussed offline - will keep SSH key in clear as it is pubic and not hidden by `show run sanitized`

```
sername cvpadmin privilege 15 secret sha512 <removed>
username cvpadmin ssh-key ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAAB[...]FBKz77PSgeMp/Rc3txFVuSmFmeTy3aMkU= cvpadmin@hostmachine.local
```

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**